### PR TITLE
feat(test): karma integration and unit test sample

### PR DESCRIPTION
**Pull request by cironunes**
/cc @IgorMinar

### CODE REVIEW

File: addon/ng2/blueprints/ng2/files/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    files: [
+      {pattern: 'node_modules/systemjs/dist/system.src.js', included: true, watched: true},
+      {pattern: 'node_modules/angular2/bundles/angular2.js', included: true, watched: true},
+      {pattern: 'node_modules/angular2/bundles/testing.js', included: true, watched: true},
+      
+      {pattern: 'karma-test-shim.js', included: true, watched: true},
+      {pattern: 'src/test/matchers.js', included: true, watched: true},
+      
+      {pattern: 'dist/**/*.js', included: false, watched: true}
+    ],
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/karma.conf.js]: what about source maps? don't you need to serve those too?

Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/karma.conf.js]: also I don't see the `proxy` configuration to enable component template fetching from tests

File: addon/ng2/blueprints/ng2/files/karma-test-shim.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+Error.stackTraceLimit = Infinity;
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000;
+
+__karma__.loaded = function() {};
+
+System.config({
+	packages: {
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/karma-test-shim.js]: can you reformat all code to indent with two spaces please.

File: addon/ng2/blueprints/ng2/files/src/test/__name___test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {describe, it, expect, beforeEachProviders, inject} from 'angular2/testing';
+import {<%= jsComponentName %>App} from '../app/<%= htmlComponentName %>';
+
+beforeEachProviders(() => [<%= jsComponentName %>App]);
+
+describe('App: KarmaIntegration', () => {
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/test/__name___test.ts]: if you want to keep this in the blueprints then it should not refer to "KarmaIntegration", you are testing the app.

Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/test/__name___test.ts]: also notice that the existing blueprints put the test files next to the source files (that's intentional) and call the files `.spec.ts`.

File: addon/ng2/blueprints/ng2/files/src/test/matchers.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+beforeEach(() => {
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/test/matchers.js]: we should move this into angular/angular/testing cli should not be in the business of distributing custom code that is not specifically needed for integration of various components.

**Comment by cironunes** [comment]
Thanks for the feedback @IgorMinar. I totally forget about the source maps. I'll add them and fix the other issues to get this merged.

Todo:

- [ ] Support source maps
- [ ] Enable component fetching
- [ ] Reformat code to 2 spaces
- [ ] Put test files next to source files
- [ ] Remove reference to `KarmaIntegration`
- [ ] Remove `matchers.js`

**Comment by cironunes** [comment]
@IgorMinar I think that all the comments were addressed. One thing that is still missing though is the ability to run the tests using `ng test` instead of `karma start`, but we can do that later IMHO.

**Comment by IgorMinar** [comment]
`ng test` can be done in a follow up pr. let me take a look at the rest.

File: addon/ng2/blueprints/ng2/files/karma-test-shim.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        .reduce(function(pathsMapping, appPath) {
+          var moduleName = appPath.replace(/^\/base\/dist\/app\//, './').replace(/\.js$/, '');
+          pathsMapping[moduleName] = appPath + '?' + window.__karma__.files[appPath]
+        return pathsMapping;
+      }, {})
+    }
+  }
+});
+
+System.import('angular2/src/core/dom/browser_adapter').then(function(browser_adapter) {
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/karma-test-shim.js]: can you add a todo comment that once beta is out we should change this code to use a "test platform".

File: addon/ng2/blueprints/ng2/files/src/app/__name__.spec.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    it('should get the meaning of life', inject([<%= jsComponentName %>App], (app) => {
+      expect(app.meaningOfLife()).toBe('The meaning of life is 42');
+      expect(app.meaningOfLife(22)).toBe('The meaning of life is 22');
+    }));
+  });
+	
+  it('should be almost passing', function() {
+    expect(1 + 1).toBe(1);
+  });
+});
```
Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/app/__name__.spec.ts]: new line please

**Comment by IgorMinar** [comment]
left a few comments. the rest looks good. can you please fix the issues and merge the PR into master? (I'll make sure you have commit rights). No merge commits please. Use rebase/fast-forward merges.

File: addon/ng2/blueprints/ng2/files/karma-test-shim.js (hunk 5 above)
Comment (cironunes) [inline addon/ng2/blueprints/ng2/files/karma-test-shim.js]: done.

File: addon/ng2/blueprints/ng2/files/src/app/__name__.spec.ts (hunk 6 above)
Comment (cironunes) [inline addon/ng2/blueprints/ng2/files/src/app/__name__.spec.ts]: done.

**Comment by cironunes** [comment]
OK. Just waiting for the commit rights.

**Comment by IgorMinar** [comment]
check email. you should have an invite.

**Comment by cironunes** [comment]
Landed as 4afd815c480d068f6aa5ac47cfa1d1d86f3c0fa7

**Pull request closed without merging**